### DISCUSSION TITLE
Creating files from index file OR toc for directories and files?

### DISCUSSION THREAD

**Discussion by Normanras** [nvim-neorg/neorg#1033] [Q&A] [answered]
I have a chance to totally revamp my writing and note-taking habits and I love the idea of Neorg, so I'm taking a deep dive to learn new habits. In my main workspace dir, I have my `index.norg`. I have a list of all my clients that I need to communicate with and am using linking/referencing to list all their notes from my meetings with them. I'm hoping to have a more efficient workflow by just creating the file once. Here are the two ways I'm approaching it and the walls I'm hitting:

1. Within the `index.norg` file, write out the referenced file name for the new file I need to write to.
```
* Company name
** {: 21.08.23 - Meeting Notes :}
```
When I save and then click enter on the referenced file, Neovim will open a new buffer that I can edit. However, when I try to save it, neovim gives me the error of: `E212: Can't open file for writing: no such file or directory`. Is this expected? Can I have Neorg create the file for me?

2. The second option here is like a table of contents but for the entire workspace/files. It works in a single file as expected, but I'd love to have every new file created auto-added and referenced in the index file with the dir structure as h2, the file as sub-points, etc. 

Before I go off and create a script to make this possible, I wanted to make sure I wasn't missing something with neorg functionality. Any tips or insights for this usecase?

**Comment by max397574** [comment, marked as answer]
I never used it before but there is the [`core.summary` module](https://github.com/nvim-neorg/neorg/wiki/Summary).
For 1. it could perhaps have something todo with the fact that this filename is perhaps wrongly escaped or something since it has so many spaces and dots and stuff
I'd try out like `21-08-23_Meeting_Notes` to see if there is a general problem with the file creation.

**Reply by Normanras** [reply]
Good point with space, I'm pretty sure my actual file names don't have spaces, but now I'll have to double check. Let me try both of your suggestions! Thanks

**Reply by Normanras** [reply]
Removal of spaces definitely solved the issue when creating the file from the index file. When creating a new file from telescope and using spaces, it creates it as expected. But there must be something with neorg creating files and not escaping the spaces, as you've said. 

Summary also works well too, I just don't like the formatting. I'll keep with the index file and creating my new files from there. Appreciate your help! Can't believe I didn't notice the spaces... 🤦🏼‍♂️

**Reply by max397574** [reply]
ig you could open an issue then for the spaces thing and also for the formatting
I'm sure it would be possible to make the format configurable

**Reply by Normanras** [reply]
It should be fairly straightforward to account for the whitespaces, so I'm going to look through the code base first and see if I can help directly. It's pretty minor but just adds that layer of accounting for additional niche uses.
